components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/ColorProcessor.kt

```diff
@@ -23,7 +24,7 @@ class ColorProcessor : IconProcessor {
         desiredSize: DesiredSize
     ): Icon {
         if (icon.color != null) return icon
-        if (request.color != null) return icon.copy(color = request.color)
+        if (request.color != null && request.color != Color.WHITE) return icon.copy(color = request.color)
```

Comment:
Feels a bit hacky (also when someone explicitly requests white in other situations.. 🤔) but okay. Let's add a comment here explaining why we do this though. :)